runtime/cudaq/algorithms/observe.h

```diff
@@ -352,6 +364,43 @@ observe_result observe(std::size_t shots, QuantumKernel &&kernel, spin_op H,
       .value();
 }
 
+/// \brief Compute the expected value of `H` with respect to `kernel(Args...)`.
+/// Specify the observation options
+template <typename QuantumKernel, typename... Args>
+  requires ObserveCallValid<QuantumKernel, Args...>
+observe_result observe(const observe_options &options, QuantumKernel &&kernel,
+                       spin_op H, Args &&...args) {
+  auto &platform = cudaq::get_platform();
+  auto kernelName = cudaq::getKernelName(kernel);
+  auto shots = options.shots;
+
+  platform.set_noise(const_cast<noise_model *>(&options.noise));
+
+  observe_result ret;
+
+  // Does this platform expose more than 1 QPU
+  // If so, let's distribute the work among the QPUs
+  if (auto nQpus = platform.num_qpus(); nQpus > 1) {
```

Comment:
This implies automatic distribution of observe measurements - I would like to make programmers specify this explicitly with the DistributionType. So for this observe overload I think we drop this segment of code and just run basic `runObservation`

Reply:
Updated in [c5d5808](https://github.com/NVIDIA/cuda-quantum/pull/664/commits/c5d58088f9dc9d6d4b143604c14fa3e445d1f88c)